org.eclipse.xtext.xbase.ui/src/org/eclipse/xtext/xbase/ui/contentassist/XbaseProposalProvider.java

```diff
@@ -786,73 +820,99 @@ protected void createLocalVariableAndImplicitProposals(
+						Iterable<JvmFeature> allFeaturesToImport = Iterables.filter(Iterables.filter(genericType.getMembers(), JvmFeature.class), new Predicate<JvmFeature>() {
 							@Override
 							public boolean apply(JvmFeature input) {
-								return !importSection.hasStaticImport(input.getSimpleName(), false);
+								boolean isValid =  !(input instanceof JvmConstructor) && input.isStatic();
+								if(isWildcard) {
+									return isValid;
+								} else {
+									return isValid && input.getSimpleName().equals(membername);
+								}
 							}
 						});
-					}
-					// Create StaticFeatureDescription instead of SimpleIdentifiableElementDescription since we want the Proposal to show parameters
-					Iterable<IEObjectDescription> scopedFeatures = Iterables.transform(featuresToImport, new Function<JvmFeature,IEObjectDescription>() {
-						@Override
-						public IEObjectDescription apply(JvmFeature feature) {
-							return new StaticFeatureDescription(QualifiedName.create(feature.getSimpleName()), feature, 0, true);
-						}
-					});
-					// Scope for all static features
-					IScope staticMemberScope = new SimpleScope(IScope.NULLSCOPE, scopedFeatures);
-					IReplacementTextApplier textApplier =  new FQNImporter(contentAssistContext.getResource(), contentAssistContext.getViewer(), staticMemberScope, qualifiedNameConverter,
-							qualifiedNameValueConverter, importSectionFactory, replaceConverter);
-					Function<IEObjectDescription, ICompletionProposal> importAddingProposalFactory = new Function<IEObjectDescription, ICompletionProposal>() {
-						@Override
-						public ICompletionProposal apply(IEObjectDescription input) {
-							ICompletionProposal proposal = proposalFactory.apply(input);
-							if(proposal instanceof ConfigurableCompletionProposal) {
-								ConfigurableCompletionProposal castedProposal = (ConfigurableCompletionProposal) proposal;
-								// Add textApplier to introduce imports if necessary
-								((ConfigurableCompletionProposal) proposal).setTextApplier(textApplier);
-								return castedProposal;
-							}
-							return proposal;
+						Iterable<JvmFeature> featuresToImport = Iterables.filter(allFeaturesToImport,filter);
+						if(context != null) {
+							// Make sure that already imported static features are not proposed
+							RewritableImportSection importSection = importSectionFactory.parse((XtextResource) context.eResource());
+							featuresToImport = Iterables.filter(featuresToImport, new Predicate<JvmFeature>() {
+								@Override
+								public boolean apply(JvmFeature input) {
+									return !importSection.hasStaticImport(input.getSimpleName(), false);
+								}
+							});
 						}
-					};
-					getCrossReferenceProposalCreator().lookupCrossReference(staticMemberScope, context, XbasePackage.Literals.XABSTRACT_FEATURE_CALL__FEATURE, acceptor,getFeatureDescriptionPredicate(contentAssistContext),importAddingProposalFactory);
+						return featuresToImport;
```

Comment:
@holgerschill this looks bogus to me. we should collect all elements and not stop on the first match

Reply:
That's a Predicate - a filter.

Reply:
featuresToImport is a list that get's filtered.

Reply:
yes but this will exit after the first element found.
i dont get any proposals

Reply:
It's fine here. Fine in the tests as well.

Reply:
Done